Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/RequestsManager.java

```diff
@@ -20,15 +29,102 @@
      * @see edu.asu.conceptpower.app.core.IRequestsManager#addReviewRequest(edu.asu.conceptpower.core.ReviewRequest)
      */
     public void addReviewRequest(ReviewRequest newReviewRequest) {
-        
+        newReviewRequest.setId(generateId(REVIEW_PREFIX));
         dbClient.store(newReviewRequest);
     }
     
     /* (non-Javadoc)
      * @see edu.asu.conceptpower.app.core.IRequestsManager#getReview(java.lang.String)
      */
-    public ReviewRequest getReview(String conceptId) {
+    public ReviewRequest getLatestReview(String conceptId) {
+        
+        ReviewRequest request = new ReviewRequest();
+        request.setConceptId(conceptId);
+        
+        List<ReviewRequest> reviewRequests = dbClient.getAllReviews(request);
+       
+        if(reviewRequests!= null && reviewRequests.size()>0) {
+            //ArrayList is initialized to make the instance mutable for sorting
+            reviewRequests = new ArrayList<>(reviewRequests);
+            Collections.sort(reviewRequests, (x, y) -> x.getCreatedAt().compareTo(y.getCreatedAt()));
+            return reviewRequests.get(reviewRequests.size() - 1);
+        }
+        
+        return null;
+    }
+    
+    /* (non-Javadoc)
+     * @see edu.asu.conceptpower.app.core.IRequestsManager#getAllReviews(java.lang.String)
+     */
+    public List<ReviewRequest> getAllReviews(String conceptId){
+        ReviewRequest request = new ReviewRequest();
+        request.setConceptId(conceptId);
+        
+        List<ReviewRequest> reviewRequests = dbClient.getAllReviews(request);
+        
+        if(reviewRequests!= null && reviewRequests.size() > 0) {
+            reviewRequests = new ArrayList<>(reviewRequests);
+            Collections.sort(reviewRequests, (x, y) -> x.getCreatedAt().compareTo(y.getCreatedAt()));
+        }
+        
+        return reviewRequests;
+    }
+    
+    /* (non-Javadoc)
+     * @see edu.asu.conceptpower.app.core.IRequestsManager#updateReview(edu.asu.conceptpower.core.ReviewRequest)
+     */
+    
+    public ReviewRequest updateReview(String reviewId, ReviewStatus reviewStatus, List<Comment> comments, OffsetDateTime createdAt,String updatedBy) {
+        ReviewRequest request = new ReviewRequest();
+        request.setId(reviewId);
+        
+        ReviewRequest storedRequest = dbClient.getReview(request);
+        
+        if(storedRequest == null) {
+            return null;
+        }
         
-        return dbClient.getReviewRequestForConcept(conceptId);
+        if(comments != null && comments.size() > 0) {
+            //There is only a possibility of single comment at a time
```

Comment:
if that's the case, why is a list being passed?

Reply:
Since I was getting in the form of ReviewRequest Object before, I had this. Removed and pushed now.